coderd/audit.go

```diff
@@ -429,6 +449,26 @@ func (api *API) auditLogResourceLink(ctx context.Context, alog database.GetAudit
 		return fmt.Sprintf("/@%s/%s/builds/%s",
 			workspaceOwner.Username, additionalFields.WorkspaceName, additionalFields.BuildNumber)
 
+	case database.ResourceTypeWorkspaceAgent:
+		if additionalFields.WorkspaceOwner != "" && additionalFields.WorkspaceName != "" {
+			return fmt.Sprintf("/@%s/%s", additionalFields.WorkspaceOwner, additionalFields.WorkspaceName)
+		}
```

Comment:
Review: I'm also a bit confused by the existing URL builder expressions. Sometimes additional fields are consulted for emptiness, then we go on to make up to 3 different queries just to fetch resources/data we already have access to.

I thought this made sense, but I'm happy to change it if requested.

Reply:
Not sure, it seems like it's called after unmarshaling additional fields so should be fine?

Reply:
I mean't "before we implemented additional fields". My bad, code wise it is all good :+1:

Reply:
Haha, now that I re-read your comment it makes much more sense. 😂 Sorry, my mind wandered elsewhere.